### PR TITLE
Feature/shriya

### PR DESCRIPTION
**Pull request by reddyrose**
# CS342 Class Application Assignment 1

## :recycle: Current situation & Problem
A class application introducing everyone in Winter 2023 CS 342 must be created. 

## :bulb: Proposed solution
A view for Shriya Reddy, with a picture and descriptions, has been added. 

## :gear: Release Notes 
A view with a profile photo, background information, hobbies, home, and LinkedIn is present. All blurbs are accessible by button. 

## :heavy_plus_sign: Additional Information
*Provide some additional information if possible*

### Related PRs
*Reference the related PRs*

### Testing
Testing for overall view is included.

### Reviewer Nudging
*Where should the reviewer start? Where is a good entry point?*

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [ ] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by reddyrose** [comment]
All tests currently pass on my local computer.

File: CS342MemberViews/Sources/ShriyaReddy/HobbyButton.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import SwiftUI
+
+private var description: String {
+        guard let descriptionPath = Bundle.module.path(forResource: "HobbyText", ofType: "txt"),
+              let description = try? String(contentsOfFile: descriptionPath) else {
+            return ""
+        }
+        
+        return description
+    }
```
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/ShriyaReddy/HobbyButton.swift]: Please add properties (including this computed properties) into the scope of a type that they are used to, e.g., the `HobbyButton` view. You can simply move it into the struct by adding it above or below the `body` computed propert.

You have correctly added the description property to the `IntroButton` in the right way.

File: CS342MemberViews/Sources/ShriyaReddy/HomeButton.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import SwiftUI
+
+
+public struct HomeButton: View {
+    @State private var showDetails = false
+    
+    @State private var region = MKCoordinateRegion(
+            center: CLLocationCoordinate2D(latitude: 44.182205, longitude: -84.506836),
+            span: MKCoordinateSpan(latitudeDelta: 10, longitudeDelta: 10)
+        )
```
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/ShriyaReddy/HomeButton.swift]: Please only indent arguments one level lower than their declaration:
```suggestion
        center: CLLocationCoordinate2D(latitude: 44.182205, longitude: -84.506836),
        span: MKCoordinateSpan(latitudeDelta: 10, longitudeDelta: 10)
    )
```

File: CS342MemberViews/Sources/ShriyaReddy/IntroButton.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @State private var showDetails = false
+
+    private var description: String {
+            guard let descriptionPath = Bundle.module.path(forResource: "IntroText", ofType: "txt"),
+                  let description = try? String(contentsOfFile: descriptionPath) else {
+                return ""
+            }
+            
+            return description
+        }
```
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/ShriyaReddy/IntroButton.swift]: Please only indent statements one level lower as the definition:
```suggestion
        guard let descriptionPath = Bundle.module.path(forResource: "IntroText", ofType: "txt"),
              let description = try? String(contentsOfFile: descriptionPath) else {
            return ""
        }
        
        return description
    }
```

File: CS342MemberViews/Sources/ShriyaReddy/ShriyaReddy.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import SwiftUI
+
+let backgroundGradient = LinearGradient(
+    colors: [
+        Color.purple,
+        Color.white
+    ],
+    startPoint: .top,
+    endPoint: .center
+)
```
Comment (PSchmiedmayer) [inline CS342MemberViews/Sources/ShriyaReddy/ShriyaReddy.swift]: Like noted in the `HobbyButton`: Please add properties (including this computed properties) into the scope of a type that they are used to, e.g., the `HobbyButton` view. You can simply move it into the struct by adding it above or below the `body` computed propert.

You have correctly added the description property to the `IntroButton` in the right way.

**Review by PSchmiedmayer — APPROVED** [review]
Very nice job @reddyrose! A nice and creative view and some cool colors 🥳

I only had a few smaller comments about the scope of properties in Swift types. Feel free to merge the PR once you have addressed the issue.

I have also merged the latest main state in your branch to make merging the PR easy as you have already resolved one merge conflict already.
Be sure that you `git pull` the latest state to your local version if you accept any of the suggestions here on GitHub and get the merge conflict to your local branch 🚀

**Comment by PSchmiedmayer** [comment]
@reddyrose A short reminder to improve your Assignment 1 submission. We would like to merge it in the next few days so you can concentrate on your project assignments 👍

Feel free to tag me if you have any questions. 
Please also tag me if you want me to take an other look at the PR.
I am also happy to help you with any merge conflicts that you might have to speed up the merge so that you can focus on the content of the exercise!

Feel free also to approach me after class today so we can take a look at the PR together and merge it directly if you have incorporated my feedback by then 🚀

**Comment by PSchmiedmayer** [comment]
Great job @reddyrose, I have merged your PR after all checks passed 👍

**Pull request merged**